data/yam/agama/auto/template.libsonnet

```diff
@@ -30,7 +32,11 @@ function(bootloader=false,
          ssl_certificates=false,
          storage='',
          user=true) {
-  [if bootloader == true then 'bootloader']: base_lib['bootloader'],
+  [if bootloader || bootloader_timeout !='' || bootloader_extra_kernel_params != '' then 'bootloader']: std.prune({
```

Comment:
the prune is unnecessary because you always put the keys, you have the if for the values.
Actually that creates a problem, that for example stopOnBootMenu doesn't  need to appear as false when is set if I get it right...
I think you can follow the structure from @lemon-suse 's PR to avoid that even if you repeat 'bootloader' of you need to make conditional the keys not the values.

Reply:
In order to add the kernel parameters, I can not totally follow Lemon's PR. Please help to review, thanks.